src/YoloDev.Expecto.TestSdk/settings.fs

```diff
@@ -14,57 +19,155 @@ type RunSettings =
+    | Filter_Test_List _ -> id
+    | Filter_Test_Case _  -> id
+    // Not applicable
+    | List_Tests -> id
+    // Not applicable
+    | Summary -> id
+    // Not applicable
+    | Summary_Location -> id
+    // Not applicable
+    | Version -> id
+    // Not applicable
+    | My_Spirit_Is_Weak -> id
+    // Not applicable, printer gets overriden in execution.fs
+    | Printer _ -> id
+    | Verbosity l -> fun o -> { o with verbosity = l }
+    // Not applicable 
+    | Append_Summary_Handler(_) -> id
+
+  let readValueParse parser elementName confNode =
+      confNode
+      |> Option.bind (Xml.element elementName)
+      |> Option.bind Xml.value
+      |> Option.bind parser
+
+  let readValueString elementName confNode = readValueParse Option.ofObj elementName confNode
+  let readValueBool elementName confNode = readValueParse TryParse.bool elementName confNode
+
+  let (|Element|_|) (name : string) (node : XElement) =
+    if name.Equals (node.Name.LocalName, StringComparison.OrdinalIgnoreCase)
+    then Some node.Value
+    else None
+
+  let (|Bool|_|) str = TryParse.bool str
+  let (|Int|_|) str = TryParse.int32 str
+  let (|Float|_|) str = TryParse.float str
+  let (|String|_|) (str : string) = Option.ofObj str
+
+  let readExpectoConfig expectoConfig (confNode: Xml.Linq.XElement) =
+    let configMapper node = 
+      match node with
+      | Element "sequenced" (Bool true) -> Some CLIArguments.Sequenced
+      | Element "sequenced" (Bool false) -> Some CLIArguments.Parallel
+      | Element "parallel" (Bool true) -> Some CLIArguments.Parallel
+      | Element "parallel" (Bool false) -> Some CLIArguments.Sequenced
+      | Element "parallel-workers" (Int i) -> Some (CLIArguments.Parallel_Workers i)
+      | Element "stress" (Float f) -> Some (CLIArguments.Stress f)
+      | Element "stress-timeout" (Float f) -> Some (CLIArguments.Stress_Timeout f)
+      | Element "stress-memory-limit" (Float f) -> Some (CLIArguments.Stress_Memory_Limit f)
+      | Element "stress-memory-limit" (Float f) -> Some (CLIArguments.Stress_Memory_Limit f)
+      | Element "fail-on-focused-tests" (Bool true) -> Some CLIArguments.Fail_On_Focused_Tests
+      | Element "debug" (Bool true) -> Some CLIArguments.Debug
+      | Element "log-name" (String s) -> Some (CLIArguments.Log_Name s)
+      | Element "filter" (String s) -> Some (CLIArguments.Filter s)
+      | Element "fscheck-max-tests" (Int i) -> Some (CLIArguments.FsCheck_Max_Tests i)
+      | Element "fscheck-start-size" (Int i) -> Some (CLIArguments.FsCheck_Max_Tests i)
+      | Element "fscheck-end-size" (Int i) -> Some (CLIArguments.FsCheck_End_Size i)
+      | Element "allow-duplicate-name" (Bool true) -> Some CLIArguments.Allow_Duplicate_Names
+      | Element "no-spinner" (Bool true) -> Some CLIArguments.No_Spinner
+      | Element "verbosity" (String s) -> Expecto.Logging.LogLevel.ofString s |> CLIArguments.Verbosity |> Some
+      | _ -> None
```

Comment:
Would it be a good idea to warn on any `Expecto.xxxx` values we don't process? I'm not sure if we have a warning channel available though...

Reply:
We can pass the Logger down into here.

```
dotnet test -- Expecto.fail-on-focused-tests=true Expecto.not-real-key=3
```

with the logger passed in and warning on it results in 

```
[Expecto 00:00:00.0660090] : Unknown config key for Expecto : not-real-key=3
```